pretyping/reductionops.ml

```diff
@@ -1083,12 +1083,17 @@ let pb_equal = function
   | Reduction.CUMUL -> Reduction.CONV
   | Reduction.CONV -> Reduction.CONV
 
+exception AnomalyInConversion of exn
+
+let _ = CErrors.register_handler (function
+    | AnomalyInConversion e ->
+      Some Pp.(str "Conversion test raised an anomaly:" ++
+               spc () ++ CErrors.print e)
+    | _ -> None)
+
 let report_anomaly (e, info) =
```

Comment:
@ppedrot @maximedenes , I remember there was some discussion as to why we absorb anomalies here, would this a good moment to add a comment ?

Reply:
The root issue is that tactics often send garbage to the kernel, e.g. asking whether two terms not having a common supertype are convertible. This breaks the conversion invariant, leading to random anomalies. I think it's indeed a good idea to write down a comment now.

Reply:
Thanks for the comment @ppedrot , a last thing before I try a comment, should we consider tactics sending rubbish to the conversion algorithm a bug in the tactics?

Reply:
I'd be inclined to believe so, but unfortunately it's not merely a uncommon behaviour. Tactics like `set` routinely break type-soundness internally, and I have no idea how to fix this without incurring a huge runtime penalty.

Reply:
Yeah that's a complex matter, not sure tho the current status is good for people writing larguish bactracking stuff.

Reply:
FTR, Looking for a head symbol as SSR does is already sufficient to rule out many nonsensical conversions. This filtering has very little runtime cost, but would clearly change the behaviour of vanilla set...